layouts/partials/nav_links.html

```diff
@@ -0,0 +1,71 @@
+<nav class="next-previous-links">
+    {{ $sectionParentPages := .CurrentSection.Parent.Pages }}
+    {{ $currentSection := .CurrentSection.Title }}
+
+
+    <sl-divider style="--spacing: 4rem 0rem;">></sl-divider>
+    <div class="nav-links">
+        <div>
+            {{ if .NextInSection }}
+                {{ with .NextInSection }}
+                    <a href="{{ .Permalink }}" class="nav-links-previous"> Previous: </a>
+                    <div>{{ .Title }}</div>
+                {{ end }}
+
+            {{ else }}
+                {{ range $sectionParentPages }}
+                    {{ if eq .Title $currentSection }}
+                        {{ with $sectionParentPages.Next . }}
+                        {{ if eq .Parent.Title "Lessons" }}
+                        <a href="{{ .Permalink }}" class="nav-links-previous">
+                            Previous lesson:</a
+                        >
+                        <div>{{ .Title }}</div>
+                        {{ else }}
```

Comment:
why are we special casing lessons?

Reply:
I thought it would be cool to say next / previous lesson, making the lesson section feel more immersive. but maybe that's not necessary

Reply:
*maybe*

However, special casing this way is complex, and coding it like this is fragile. I'd vote to just keep it simple, otherwise, infer the type of the next section from some variable or front-matter.

e.g. `.Parent.series` or custom front-matter section `.Parent.Params.section_name` https://gohugo.io/content-management/front-matter/